src/interactive-window/editor-integration/codeLensFactory.ts

```diff
@@ -64,10 +63,7 @@ export class CodeLensFactory implements ICodeLensFactory {
         notebook.onDidChangeNotebookCellExecutionState(this.onDidChangeNotebookCellExecutionState, this, disposables);
         kernelProvider.onDidDisposeKernel(
             (kernel) => {
-                const notebook = getAssociatedNotebookDocument(kernel);
-                if (notebook) {
-                    this.notebookData.delete(notebook.uri.toString());
-                }
+                this.notebookData.delete(kernel.notebook.uri.toString());
```

Comment:
Maybe out of scope for this PR, but I thought we removed almost all cases of using uri.toString(). Seems like this should maybe WeakMap the NotebookDocument instead like we do in other places.